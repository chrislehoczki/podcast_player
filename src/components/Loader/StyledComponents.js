import styled from "styled-components";

export const Ring = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;
