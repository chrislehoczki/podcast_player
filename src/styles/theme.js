import React from "react";
import styled from "styled-components";

const theme = {
  primary: "#ff7426",
  background: "#3e3e3e"
};

const Loader = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;

const PlayButton = (props, onClick) => {
  if (!props.loaded) {
    return (
      <LoaderContainer size={props.size}>
        <Loader />
      </LoaderContainer>
    );
  }
  return <Button {...props} onClick={onClick} />;
};

const Button = styled.button`
  background: url(${props => props.img}) no-repeat center center;
  background-size: 32px;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  filter: invert(1);
  cursor: pointer;
  border-radius: 100%;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;

export default theme;

console.log(PlayButton);
export { Button, PlayButton };
