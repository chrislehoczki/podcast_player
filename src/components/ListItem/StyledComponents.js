import styled from "styled-components";
import theme from "../../styles/theme";

export const ListContainer = styled.div`
  display: flex;
  padding: 10px;
  border: 1px solid ${theme.primary};
`;

export const DetailContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
  width: 400px;
`;

export const MainDetail = styled.h4`
  margin: 0;
`;

export const SubDetail = styled.div`
  margin: 0;
  font-size: 14px;
`;

export const Duration = styled.p`
  margin: 3px 0;
  font-size: 14px;
`;
