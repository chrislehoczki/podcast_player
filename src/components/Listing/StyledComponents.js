import styled from "styled-components";
import theme from "../../styles/theme";

export const Container = styled.div`
  width: 100%;
  height: calc(100vh - 40px - 400px);
  overflow-y: auto;
  ::-webkit-scrollbar {
    width: 10px;
  }
  ::-webkit-scrollbar-track {
    background: 0;
  }
  ::-webkit-scrollbar-thumb {
    background: ${theme.primary};
  }
  @media only screen and (max-width: 400px) {
    height: calc(100vh - 400px);
  }
`;
