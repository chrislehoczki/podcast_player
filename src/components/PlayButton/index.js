import React from "react";
import PropTypes from "prop-types";
import { Button } from "../../styles/theme";
import Loader from "../Loader";

const PlayButton = ({
  loaded,
  playing,
  stopPodcast,
  playPodcast,
  podcastId
}) => {
  return !loaded ? (
    <Loader size={60} />
  ) : playing ? (
    <Button
      img={"icons/pause.png"}
      size={60}
      border={true}
      onClick={() => loaded && stopPodcast()}
    />
  ) : (
    <Button
      img={"icons/play.png"}
      size={60}
      border={true}
      onClick={() => playPodcast({ id: podcastId })}
    />
  );
};

PlayButton.propTypes = {
  playPodcast: PropTypes.func,
  stopPodcast: PropTypes.func,
  loaded: PropTypes.bool,
  playing: PropTypes.bool,
  podcastId: PropTypes.string
};

export default PlayButton;
