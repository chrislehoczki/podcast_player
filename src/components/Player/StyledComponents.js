import styled from "styled-components";
import theme from "../../styles/theme";

export const PlayerContainer = styled.div`
  width: 100%;
  height: 400px;
  position: relative;
  background: ${theme.background};
  border-bottom: 2px solid ${theme.primary};
`;

export const Image = styled.div`
  width: 150px;
  height: 150px;
  min-height: 150px;
  background: url(${props => props.img}) no-repeat center center;
  background-size: cover;
`;

export const Details = styled.div`
  height: 60%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  padding: 20px 10px;
`;

export const Title = styled.h1`
  margin: 0;
  font-size: 18px;
  margin: 20px 0 5px 0;
  line-height: 22px;
`;

export const Author = styled.h2`
  font-size: 16px;
  margin: 0;
`;

export const Controls = styled.div`
  display: flex;
  vertical-align: center;
  align-items: center;
  width: 100%;
  height: 20%;
  justify-content: center;
`;
