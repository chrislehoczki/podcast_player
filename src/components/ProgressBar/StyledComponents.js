import styled from "styled-components";
import theme from "../../styles/theme";

const sliderThumbStyles = props => `
  width: 15px;
  height: 15px;
  border-radius: 100%;
  background: ${theme.primary};
`;

export const Container = styled.div`
  position: absolute;
  bottom: 25px;
  width: 80%;
  margin: 0;
  display: flex;
  align-items: center;
  color: ${theme.primary};
  .slider {
    flex: 6;
    -webkit-appearance: none;
    width: 100%;
    height: 5px;
    border-radius: 5px;
    background: ${theme.primary};
    outline: none;
    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      ${props => sliderThumbStyles(props)}
    }
    &::-moz-range-thumb {
      ${props => sliderThumbStyles(props)}
    }
  }
`;
