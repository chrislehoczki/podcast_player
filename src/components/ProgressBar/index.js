import React, { useEffect, useState } from "react";
import podcastPlayer from "../../lib/PodcastPlayer";
import { Container } from "./StyledComponents";

const ProgressBar = () => {
  const [progress, setProgress] = useState(0);
  useEffect(() => {
    podcastPlayer.createProgressInterval(progress => {
      setProgress(+progress);
    });
  }, [setProgress]);
  return (
    <Container>
      <input
        type="range"
        min={0}
        max={1}
        value={progress || 0}
        step={0.001}
        className="slider"
        readOnly={true}
      />
    </Container>
  );
};

export default ProgressBar;
