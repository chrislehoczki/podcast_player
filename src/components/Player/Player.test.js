import React from "react";
import { shallow, mount } from "enzyme";
import mockData from "../../test/mock_data";
import { Image, Title, Author } from "./StyledComponents";
import PlayButton from "../PlayButton";
import ProgressBar from "../ProgressBar";
import { Button } from "../../styles/theme";
import { Player } from "./index";
const {
  title,
  author,
  artwork: {
    urls: [{ url: img }]
  }
} = mockData[0];

describe("<Player />", () => {
  it("Renders correctly", () => {
    const component = shallow(<Player />);
    expect(component).toMatchSnapshot();
  });
  it("Renders a play button", () => {
    const component = shallow(<Player />);
    expect(component.find(PlayButton)).toExist();
  });
  it("Renders a progress bar", () => {
    const component = shallow(<Player />);
    expect(component.find(ProgressBar)).toExist();
  });
  it("Renders a next and previous button", () => {
    const component = mount(<Player />);
    expect(component.find(Button).first()).toHaveStyleRule(
      "background",
      `url(${"icons/prev.png"}) no-repeat center center`
    );
    expect(component.find(Button).last()).toHaveStyleRule(
      "background",
      `url(${"icons/next.png"}) no-repeat center center`
    );
  });
  it("Renders data fields correctly", () => {
    const component = mount(<Player title={title} author={author} img={img} />);
    expect(component.find(Title).text()).toEqual(title);
    expect(component.find(Author).text()).toEqual(author);
    expect(component.find(Image)).toHaveStyleRule(
      "background",
      `url(${img}) no-repeat center center`
    );
  });
});
