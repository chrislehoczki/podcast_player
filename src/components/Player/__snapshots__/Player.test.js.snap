// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> Renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div />
    <styled.h1 />
    <styled.h2 />
  </styled.div>
  <styled.div>
    <styled.button
      img="icons/prev.png"
      onClick={[Function]}
      size={40}
    />
    <PlayButton />
    <styled.button
      img="icons/next.png"
      onClick={[Function]}
      size={40}
    />
    <ProgressBar />
  </styled.div>
</styled.div>
`;
